bugsnag-android-core/src/main/java/com/bugsnag/android/MetadataState.kt

```diff
@@ -0,0 +1,35 @@
+package com.bugsnag.android
+
+internal data class MetadataState(val metadata: Metadata = Metadata()) : BaseObservable(),
+    MetadataAware {
+
+    override fun addMetadata(section: String, value: Map<String, Any?>) = metadata.addMetadata(section, value)
+    override fun addMetadata(section: String, key: String, value: Any?) {
+        metadata.addMetadata(section, key, value)
+
+        when (value) {
+            null -> notifyClear(section, key)
+            else -> notifyObservers(StateEvent.AddMetadata(section, key, metadata.getMetadata(section, key)))
+        }
+    }
+
+    override fun clearMetadata(section: String) {
+        metadata.clearMetadata(section)
+        notifyClear(section, null)
+        notifyObservers(StateEvent.ClearMetadataTab(section))
```

Comment:
This appears to be a duplicate call to `notifyObservers` - it's already called through `notifyClear`.

Reply:
Good catch - I've removed the duplicate call.